src/elements/content-sidebar/BoxAISidebarContent.tsx

```diff
@@ -83,32 +84,39 @@ function BoxAISidebarContent(props: ApiWrapperWithInjectedProps & { shouldShowLa
         setCacheValue('encodedSession', encodedSession);
     }
 
-    if (cache.questions !== questions) {
+    if (!isEqual(cache.questions, questions)) {
         setCacheValue('questions', questions);
     }
 
-    if (cache.agents.selectedAgent !== selectedAgent) {
+    if (
+        !isEqual(cache.agents?.selectedAgent, selectedAgent) ||
+        !isEqual(cache.agents?.agents, agents) ||
+        !isEqual(cache.agents?.requestState, requestState)
+    ) {
```

Comment:
i think you can simplify this:

```suggestion
    if (!isEqual(cache.agents, { agents, requestState, selectedAgent })) {
```

or

```
const agentState = useAgents();
const { agents, requestState, selectedAgent } = agentState;

...

if (!isEqual(cache.agents, agentState)) {
```